posthog/models/event.py

```diff
@@ -237,35 +246,58 @@ def person(self):
     # the event won't be in the Action-Event relationship yet.
     @property
     def actions(self) -> List:
+        last_updated_action_ts = Action.objects.filter(team_id=self.team_id).aggregate(models.Max("updated_at"))['updated_at__max']
+
         actions = (
             Action.objects.filter(
                 team_id=self.team_id,
-                steps__event=self.event, # filter by event name to narrow down
+                steps__event=self.event,  # filter by event name to narrow down
                 deleted=False
             )
-            .distinct("id")
-            .prefetch_related(
+                .distinct("id")
+                .prefetch_related(
                 Prefetch("steps", queryset=ActionStep.objects.order_by("id"))
             )
         )
-        if len(actions) == 0:
-            return []
-        events: models.QuerySet[Any] = Event.objects.filter(pk=self.pk)
-        for action in actions:
-            events = events.annotate(
-                **{
-                    "action_{}".format(action.pk): Event.objects.filter(pk=self.pk)
-                    .query_db_by_action(action)
-                    .values("id")[:1]
-                }
-            )
-        event = [event for event in events][0]
+        if not self.team_id in LAST_UPDATED_TEAM_ACTION \
+            or not self.team_id in TEAM_EVENT_ACTION_QUERY_CACHE \
+            or not self.event in TEAM_EVENT_ACTION_QUERY_CACHE[self.team_id] \
+            or not self.team_id in TEAM_ACTION_QUERY_CACHE \
+            or (last_updated_action_ts > LAST_UPDATED_TEAM_ACTION[self.team_id]):
```

Comment:
Is this logic implicitly tested by the current tests, or should we add another test to test this logic specifically? I think we do have some tests that test updating of actions, just want to make sure they would hit this. Maybe a specific test would be good?

Reply:
I think a specific test would be the right way to do this. That's what I was going to hack on a little later if I have time.